kombu/utils/compat.py

```diff
@@ -7,7 +7,11 @@
 from functools import wraps
 
 from contextlib import contextmanager
-import importlib_metadata
+
+try:
+    from importlib import metadata as importlib_metadata
+except ImportError:
+    import importlib_metadata
```

Comment:
This line is still going to cause an issue for eventlet and therefore open stack folks (because they use eventlet).  Celery [does support eventlet](https://docs.celeryproject.org/en/latest/userguide/concurrency/eventlet.html).  Concurrent thread:  https://github.com/celery/kombu/issues/1082

Reply:
good catch Matt

Reply:
Actually though -- if we ensure eventlet users try the fix that @davidszotten mentioned "eventlet 0.25.1 has been released with a fix for the pathlib issue" maybe this code would be safer.